opcua/common/events.py

```diff
@@ -19,19 +24,66 @@ def get_event_props_as_fields_dict(self):
         """
         field_vars = {}
         for key, value in vars(self).items():
-            if not key.startswith("__") and key is not "server_handle":
-                field_vars[key] = ua.Variant(value)
+            if not key.startswith("__") and key not in self.internal_properties:
+                field_vars[key] = ua.Variant(value, self.data_types[key])
         return field_vars
 
+    @staticmethod
+    def from_field_dict(fields):
+        """
+        Create an Event object from a dict of name and variants
+        """
+        result = EventResult()
+        for k, v in fields.items():
+            setattr(result, k, v.Value)
+            result.data_types[k] = v.VariantType
+        return result
```

Comment:
@zerox1212 this method should make it easy for you to create an event and return it in read_event_history